app/Providers/RouteServiceProvider.php

```diff
@@ -19,19 +19,25 @@
 use App\Helpers\IdHasher;
 use Illuminate\Routing\Router;
 use Illuminate\Support\Facades\Route;
+use Illuminate\Console\DetectsApplicationNamespace;
 use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Illuminate\Foundation\Support\Providers\RouteServiceProvider as ServiceProvider;
 
 class RouteServiceProvider extends ServiceProvider
 {
+    use DetectsApplicationNamespace;
+
     /**
      * This namespace is applied to your controller routes.
      *
      * In addition, it is set as the URL generator's root namespace.
      *
      * @var string
      */
-    protected $namespace = 'App\Http\Controllers';
+    protected function getNamespace()
```

Comment:
I don't understand this part of having to change namespace. Can you explain?

Reply:
It uses the trait `DetectsApplicationNamespace` from Laravel to get the root namespace, which I think is nice.